pixelgl/input.go

```diff
@@ -408,7 +410,24 @@ func (w *Window) UpdateInput() {
 	mainthread.Call(func() {
 		glfw.PollEvents()
 	})
+	w.doUpdateInput()
+}
+
+// WaitInput blocks until an event is received or a timeout. If timeout is 0
+// then it will wait indefinitely
+func (w *Window) WaitInput(timeout time.Duration) {
+	mainthread.Call(func() {
+		if timeout == 0 {
+			glfw.WaitEvents()
+		} else {
+			glfw.WaitEventsTimeout(timeout.Seconds())
+		}
+	})
+	w.doUpdateInput()
+}
 
+// internal input bookkeeping
+func (w *Window) doUpdateInput() {
```

Comment:
This would be the first unexported method in this file. Is there a way to name this in such a way that communicates its intention (admittedly I don't totally understand its intension) and also make it exported so that it is in line with the rest of the methods?

Reply:
This code block is responsible for tracking things link: `Pressed` vs `JustPressed`.

It doesn't make sense to export this because its semantic meaning is valid only after a call to either `UpdateInput` or `WaitEvent`.

The alternative would be `code-duplication`. This is similar to some Go std libraries like (https://github.com/golang/go/blob/go1.14.4/src/sync/once.go#L40)

Reply:
Awesome - thanks so much for the explanation and education!